src/common/transformations/src/transformations/common_optimizations/matmul_split_decomposition.cpp

```diff
@@ -0,0 +1,214 @@
+        std::shared_ptr<ov::Node> add = nullptr;
+        bool have_bias = false;
+        for (auto& consumer : matmul->get_output_target_inputs(0)) {
+            if (ov::is_type<opset1::Add>(consumer.get_node()->shared_from_this())) {
+                add = pattern_map.at(add_pattern).get_node_shared_ptr();
+                have_bias = true;
+                break;
+            }
+        }
+        const bool& transpose_b = as_type_ptr<opset1::MatMul>(matmul)->get_transpose_b();
+        const auto& reshape = pattern_map.at(reshape_pattern);
+        auto concat = reshape.get_node_shared_ptr()->input_value(1);
+
+        auto reshape_output_shape_rank = reshape.get_node_shared_ptr()->get_output_partial_shape(0).rank().get_length();
+        if (reshape_output_shape_rank != 5) {
+            // Baichun also match this pattern, but it only has rank 4, and have performance regression, so filter it
+            // out.
+            return false;
+        }
+
+        NodeVector gathers, fake_quantizes;
+        gathers.resize(3);
+        fake_quantizes.resize(3);
+        for (auto& child : children) {
+            std::shared_ptr<ov::Node> fq = nullptr;
+            auto gather = child.get_node()->shared_from_this();
+            if (ov::is_type<opset1::FakeQuantize>(gather)) {
+                fq = gather;
+                gather = gather->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+            }
+            if (ov::is_type<ov::op::util::GatherBase>(gather)) {
+                const auto axis_node = as_type_ptr<opset6::Constant>(gather->input_value(2).get_node_shared_ptr());
+                if (axis_node) {
+                    const auto& axis_val = axis_node->cast_vector<int32_t>();
+                    if (axis_val.size() != 1u || axis_val[0] != 0) {
+                        return false;
+                    }
+                } else {
+                    return false;
+                }
+
+                const auto indices_node = as_type_ptr<opset6::Constant>(gather->input_value(1).get_node_shared_ptr());
+                if (indices_node) {
+                    const auto& indices_val = indices_node->cast_vector<int32_t>();
+                    if (indices_val.size() != 1u) {
+                        return false;
+                    }
+                    if (indices_val[0] < 0 || indices_val[0] >= 3) {
+                        return false;
+                    }
+                    gathers[indices_val[0]] = gather;
+                    fake_quantizes[indices_val[0]] = fq;
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+
+        if (std::any_of(gathers.begin(), gathers.end(), [](const std::shared_ptr<Node> node_ptr) {
+                return !node_ptr || !is_type<ov::op::util::GatherBase>(node_ptr);
+            })) {
+            return false;
+        }
```

Comment:
It looks like unnecessary check: we already check that while filled the `gathers` vector